parameters/src/testnet3/mod.rs

```diff
@@ -18,74 +18,239 @@ pub mod genesis;
-impl_remote!(ShiftedDegree24, REMOTE_URL, "resources/", "shifted-powers-of-beta-24", "usrs");
-impl_remote!(ShiftedDegree25, REMOTE_URL, "resources/", "shifted-powers-of-beta-25", "usrs");
-impl_remote!(ShiftedDegree26, REMOTE_URL, "resources/", "shifted-powers-of-beta-26", "usrs");
-impl_remote!(ShiftedDegree27, REMOTE_URL, "resources/", "shifted-powers-of-beta-27", "usrs");
+// impl_local!(ShiftedDegree15, "resources/", "shifted-powers-of-beta-15", "usrs");
+// impl_remote!(ShiftedDegree16, REMOTE_URL, "resources/", "shifted-powers-of-beta-16", "usrs");
+// impl_remote!(ShiftedDegree17, REMOTE_URL, "resources/", "shifted-powers-of-beta-17", "usrs");
+// impl_remote!(ShiftedDegree18, REMOTE_URL, "resources/", "shifted-powers-of-beta-18", "usrs");
+// impl_remote!(ShiftedDegree19, REMOTE_URL, "resources/", "shifted-powers-of-beta-19", "usrs");
+// impl_remote!(ShiftedDegree20, REMOTE_URL, "resources/", "shifted-powers-of-beta-20", "usrs");
+// impl_remote!(ShiftedDegree21, REMOTE_URL, "resources/", "shifted-powers-of-beta-21", "usrs");
+// impl_remote!(ShiftedDegree22, REMOTE_URL, "resources/", "shifted-powers-of-beta-22", "usrs");
+// impl_remote!(ShiftedDegree23, REMOTE_URL, "resources/", "shifted-powers-of-beta-23", "usrs");
+// impl_remote!(ShiftedDegree24, REMOTE_URL, "resources/", "shifted-powers-of-beta-24", "usrs");
+// impl_remote!(ShiftedDegree25, REMOTE_URL, "resources/", "shifted-powers-of-beta-25", "usrs");
+// impl_remote!(ShiftedDegree26, REMOTE_URL, "resources/", "shifted-powers-of-beta-26", "usrs");
+// impl_remote!(ShiftedDegree27, REMOTE_URL, "resources/", "shifted-powers-of-beta-27", "usrs");
+
+// // Powers of Beta Times Gamma * G
+// impl_local!(Gamma, "resources/", "powers-of-beta-gamma", "usrs");
+// // Negative Powers of Beta in G2
+// impl_local!(NegBeta, "resources/", "neg-powers-of-beta", "usrs");
+// // Negative Powers of Beta in G2
+// impl_local!(BetaH, "resources/", "beta-h", "usrs");
+
+// // Mint
+// impl_remote!(MintProver, REMOTE_URL, "resources/", "mint", "prover");
+// impl_remote!(MintVerifier, REMOTE_URL, "resources/", "mint", "verifier");
+// // Transfer
+// impl_remote!(TransferProver, REMOTE_URL, "resources/", "transfer", "prover");
+// impl_remote!(TransferVerifier, REMOTE_URL, "resources/", "transfer", "verifier");
+// // Join
+// impl_remote!(JoinProver, REMOTE_URL, "resources/", "join", "prover");
+// impl_remote!(JoinVerifier, REMOTE_URL, "resources/", "join", "verifier");
+// // Split
+// impl_remote!(SplitProver, REMOTE_URL, "resources/", "split", "prover");
+// impl_remote!(SplitVerifier, REMOTE_URL, "resources/", "split", "verifier");
+// // Fee
+// impl_remote!(FeeProver, REMOTE_URL, "resources/", "fee", "prover");
+// impl_remote!(FeeVerifier, REMOTE_URL, "resources/", "fee", "verifier");
+
+use std::sync::Mutex;
+
+lazy_static! {
+    pub static ref PARAMETER_PROVIDER: Mutex<IndexMap<String, Vec<u8>>> = {
```

Comment:
This is the heart of the change. We now have a `PARAMETER_PROVIDER` exposed to the WASM so it can upload files as necessary. We initialize empty Vecs for the rest of the.